### ISSUE TITLE
Underwater curtain begins glitching as distance from world center increases (and when viewed from a steep angle)

### ISSUE THREAD

**Issue by Nimuecrypto** [huwb/crest-oceanrender#183]
This will probably be an easy fix since it's likely very similar to the most recent underwater curtain issue you resolved.  

Here is a video showing the issue:

https://www.youtube.com/watch?v=bz9WVLInW6s&feature=youtu.be

**Comment by huwb** [comment]
It's weird, i wasnt really able to repro this. At (100,000, 0, 100,000) i see a slight flicker, but nothing at 50,000. I parented a sphere to the camera and looked straight up and straight down.

Some instability is inevitable due to precision issues (games i worked on in the past stayed within 4000 units from teh origin) so I dont expect this to work perfectly further out. Still its odd that the issues seem so much more severe in your tests. I'll have a think about what the difference could be.

**Comment by huwb** [comment]
Oh also could you let me know which commit or version you were running if you have this info.

**Comment by Nimuecrypto** [comment]
@huwb I was using a version that was a couple weeks old, but after seeing your comment I downloaded the version you've been testing on.

Result is exactly the same as before the update.  Yeah I don't know why you can't repro.

As you said, some of the test distances are pretty ridiculous.. but I start seeing noticable degradation of the shader on my machine at far smaller distances (<10000).   

When I get a moment I'll see if I can do further testing to see if I can discover what may be causing my results to differ from yours.

In case the difference turns out to be hardware related,  I am using a dual-monitor setup with both monitors connected to a Radeon R9 290X, running driver version 18.12.1.1

**Comment by huwb** [comment]
Ok let me know.

Yeah it could be hardware dependent, maybe some kind of precision issue. My ability to track down hardware dependent issues that i cant repro is severely limited, as I dont have the resources to do more.

On previous projects that i've worked on we had to limit the playable area to around 4km squared. Other things like physics would also become unstable beyond this kind of distance. Is limiting your playable area an option? Alternatively, we recently added floating origin functionality to avoid large distances from the origin, is this an option?
